### PR TITLE
Terminal history now shows invalid commands. (#351)

### LINKED ISSUE #351: Terminal history display only valid commands
**Describe the bug**
Only valid commands are displayed in the history.

**Reproduce**
Unbenannt

### PR DESCRIPTION
**Pull request by Tristan-H11**
**Issue**
Closes #351

### CODE REVIEW

File: src/app/desktop/windows/terminal/terminal-states.ts [hunk 1]
```diff
@@ -39,27 +39,25 @@ export abstract class CommandTerminalState implements TerminalState {
...
-        }
+      // Flagged commands won't be logged.
+      if (this.commands[command].hideFromProtocol) {
+        return;
       }
-    } else if (command !== '') {
+    } else {
       this.commandNotFound(command);
     }
+    this.protocol.unshift((command + ' ' + args.join(' ')).trim());
```
Comment (MarcelCoding) [inline src/app/desktop/windows/terminal/terminal-states.ts:55]:
```suggestion
    this.protocol.unshift(`${command} ${args.join(' ')`);
```

File: src/app/desktop/windows/terminal/terminal-states.ts [hunk 2]
```diff
@@ -39,27 +39,25 @@ export abstract class CommandTerminalState implements TerminalState {
...
   }
 
   execute(command: string) {
-    const command_ = command.trim().split(' ');
-    if (command_.length === 0) {
+    const command_ = command.trim().split(' ').filter(e => e.length);
```
Comment (MarcelCoding) [inline src/app/desktop/windows/terminal/terminal-states.ts:59]:
```suggestion
    const command_ = command.split(' ').map(e => e.trim()).filter(e => e.length);
```

**Review by MarcelCoding — CHANGES_REQUESTED** [review]

File: src/app/desktop/windows/terminal/terminal-states.ts (hunk 1 above)
Comment (Tristan-H11) [reply src/app/desktop/windows/terminal/terminal-states.ts:55]: This way, a command without args would be saved as `'command '`. Thus, the `.trim()` is needed

File: src/app/desktop/windows/terminal/terminal-states.ts (hunk 2 above)
Comment (Tristan-H11) [inline src/app/desktop/windows/terminal/terminal-states.ts:59]:
```suggestion
    const command_ = command.split(' ').filter(e => e.length);
```

**Review by MarcelCoding — APPROVED** [review]

**Review by Defelo — APPROVED** [review]

**Pull request merged**